lib/Doctrine/DBAL/Driver/Ibase/AbstractIbaseConnection.php

```diff
@@ -0,0 +1,500 @@
+<?php
+
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Driver\Ibase;
+
+use Doctrine\DBAL\Driver\Connection;
+use Doctrine\DBAL\Driver\ServerInfoAwareConnection;
+use Doctrine\DBAL\Platforms\FirebirdPlatform;
+
+/**
+ * ibase-api implementation of the Connection interface.
+ * 
+ * <b>This Driver/Platform is in Beta state</b>
+ * 
+ * @author Andreas Prucha, Helicon Software Development <prucha@helicon.co.at>
+ */
+abstract class AbstractIbaseConnection implements Connection, ServerInfoAwareConnection
+{
+
+    /**
+     * Attribute to set the default transaction isolation level.
+     * 
+     * @see \Doctrine\DBAL\Connection::TRANSACTION_READ_COMMITTED
+     * @see \Doctrine\DBAL\Connection::TRANSACTION_READ_UNCOMMITTED
+     * @see \Doctrine\DBAL\Connection::TRANSACTION_REPEATABLE_READ
+     * @see \Doctrine\DBAL\Connection::TRANSACTION_SERIALIZABLE
+     * 
+     */
+    const ATTR_DOCTRINE_DEFAULT_TRANS_ISOLATION_LEVEL = 'doctrineTransactionIsolationLevel';
+
+    /**
+     * Transaction wait timeout in case of an locking conflict
+     */
+    const ATTR_DOCTRINE_DEFAULT_TRANS_WAIT = 'doctrineTransactionWait';
+
+    /**
+     * @var string Full database identifier passed to ibase_connect
+     */
+    protected $dbs;
+
+    /**
+     * @var string Connection Username
+     */
+    protected $username;
+
+    /**
+     * @var string Connection Password
+     */
+    protected $password;
+
+    /**
+     * @var string Charset
+     */
+    protected $charset = null;
```

Comment:
Any reason why you save dbs, username, password and charset on the connection object?